File: OOP_Python_Tiền_Điện/receipt.py
from client import Client

class Receipt(Client):
    def __init__(self, listCustomers):
        super().enterInfo(listCustomers)
        self.enterElectricIndex()
    def getMoneyToBePaid(self):
        return self.moneyToBePaid
    def setOldIndex(self,oldIndex):
        self.oldIndex = oldIndex
    def setNewIndex(self,newIndex):
        self.newIndex = newIndex
    def enterElectricIndex(self):
        while True:
            oldIndex = int(input("Nhập chỉ số đầu: "))
            newIndex = int(input("Nhập chỉ số cuối:"))
            if oldIndex < 0 or newIndex < 0: 
                super().drawLine()
                print("Chỉ số không hợp lệ")
                super().drawLine()
            else:
                if oldIndex < newIndex:
                    self.setOldIndex(oldIndex)
                    self.setNewIndex(newIndex)
                    break
                else:
                    super().drawLine()
                    print("Lỗi chỉ số đầu lớn hơn chỉ số cuối !!!")
                    super().drawLine()
        self.moneyToBePaid= (self.newIndex - self.oldIndex) * 750
    def showInfo(self):
        super().drawLine()
        super().showInfo();
        print(f"Chỉ số đầu: {self.oldIndex}")
        print(f"Chỉ số cuối: {self.newIndex}")
        print(f"->Số tiền trả: {self.moneyToBePaid} VNĐ")
        super().drawLine()
    

File: OOP_Java_Tiền_Điện/Receipt.java
import java.util.ArrayList;
import java.util.InputMismatchException;
import java.util.Scanner;

public class Receipt extends Client {
    private int oldIndex, newIndex, moneyToBePaid ;
    public Receipt(ArrayList<Client> clients){
        this.enterInfo(clients);
    }
    public void enterInfo(ArrayList<Client> cn){
        Scanner input = new Scanner(System.in);
        super.enterInfo(cn);
        int error = 0;
        while(true){
            System.out.print("Nhập chỉ số điện cũ: ");
            String oldIndex = input.nextLine().trim();
            if(!oldIndex.isBlank()&&super.isNumeric(oldIndex)){
                this.oldIndex = Integer.parseInt(oldIndex);
                break;
            } else {
                System.out.println("Dữ liệu nhập vào không hợp lệ !!! Vui lòng nhập lại.");
                if (error == 2){
                    System.out.println("Có vẻ đã có lỗi nhập liệu tại đây.Hệ thống sẽ được đóng.");
                    throw new InputMismatchException("Oops, lỗi nhập dữ liệu.");
                }
                error++;
                DrawLine();
            }
        }
        while(true){
            System.out.print("Nhập chỉ số điện mới: ");
            String newIndex = input.nextLine().trim();
            if(!newIndex.isBlank()&&super.isNumeric(newIndex)){
                if(Integer.parseInt(newIndex) >= this.oldIndex){
                    this.newIndex = Integer.parseInt(newIndex);
                    break;
                } else  {
                    DrawLine();
                    System.out.println("Chỉ số điện mới không hợp lệ. Vui lòng nhập lại");
                    DrawLine();
                }
            } else {
                System.out.println("Dữ liệu nhập vào không hợp lệ !!! Vui lòng nhập lại.");
                if (error == 4){
                    System.out.println("Có vẻ đã có lỗi nhập liệu tại đây.Hệ thống sẽ được đóng.");
                    throw new InputMismatchException("Oops, lỗi nhập dữ liệu.");
                }
                error++;
                DrawLine();
            }
        }
        this.moneyToBePaid = (newIndex - oldIndex) * 750;
        DrawLine();
    }
    public void showInfo(){
        super.showInfo();
        System.out.println("Chỉ số điện đầu: " + this.oldIndex + "| Chỉ số điện cuối: " + this.newIndex);
        System.out.println("Tiền điện phải trả: " + this.moneyToBePaid + "$");
        DrawLine();
    }

}
